pkg/cmd/webhooks/forward.go

```diff
@@ -0,0 +1,253 @@
+		HttpClient: f.HttpClient,
+		IO:         f.IOStreams,
+		Config:     f.Config,
+	}
+	cmd := cobra.Command{
+		Use:   "forward --event=<event_type> --repo=<repo> --port=<port> [--host=<host>]",
+		Short: "Receive test webhooks locally",
+		Example: heredoc.Doc(`
+			# create a dev webhook for the 'issue_open' event in the monalisa/smile repo in GitHub running locally, and
+			# forward payloads for the triggered event to localhost:9999
+
+			$ gh webhooks forward --event=issue_open --repo=monalisa/smile --port=9999 --host=api.github.localhost
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if opts.EventType == "" {
+				return cmdutil.FlagErrorf("`--event` flag required")
+			}
+			if opts.Repo == "" {
+				return cmdutil.FlagErrorf("`--repo` flag required")
+			}
+			if opts.Host == "" {
+				fmt.Fprintf(opts.IO.Out, "No --host specified, connecting to default GitHub.com \n")
+				opts.Host = gitHubAPIProdURL
+			}
+
+			if opts.Port == 0 {
+				fmt.Fprintf(opts.IO.Out, "No --port specified, printing webhook payloads to stdout \n")
+			}
+
+			if runF != nil {
+				return runF(&opts)
+			}
+
+			wsURLString, err := createHook(&opts)
+			if err != nil {
+				return err
+			}
+
+			wsURL, err := url.Parse(wsURLString)
+			if err != nil {
+				return err
+			}
+
+			cfg, err := opts.Config()
+			if err != nil {
+				return err
+			}
+			token, _ := cfg.AuthToken(opts.Host)
+
+			var retries int
+			events := make(chan wsEventReceived)
+			replies := make(chan httpEventForward)
+
+			go forwardEvents(&opts, events, replies)
+
+			for {
+				if retries >= maxRetries {
+					fmt.Fprintf(opts.IO.Out, "Unable to connect to webhooks server, forwarding stopped.\n")
+					return nil
+				}
+
+				done := make(chan struct{})
+				defer close(done)
+				conn, err := handleWebsocket(token, wsURL, &opts, events)
+
+				if err != nil {
+					unwrapped := errors.Unwrap(err)
+					var syscallErr *os.SyscallError
+					// If the error is a TCP handleWebsocket error, retry connecting
+					if errors.As(unwrapped, &syscallErr) && (unwrapped.Error() == "connect: connection refused") {
+						retries += 1
+						time.Sleep(5 * time.Second)
+						continue
+					}
+					// If the error is a server disconnect (1006), retry connecting
+					if websocket.IsCloseError(err, websocket.CloseAbnormalClosure) {
+						retries += 1
+						time.Sleep(10 * time.Second)
+						continue
+					}
+
+					if websocket.IsCloseError(err, websocket.CloseNormalClosure) {
+						return nil
+					}
+					return err
+				}
+
+				fmt.Fprintf(opts.IO.Out, "Connected to webhooks server, forwarding \n")
+				retries = 0
+
+				go func() {
+					for {
+						select {
+						case <-done:
+							return
+						case reply := <-replies:
+							err := conn.WriteJSON(reply)
+							if err != nil {
+								return
+							}
+						}
+					}
+				}()
+			}
+			return nil
+		},
+	}
+	cmd.Flags().StringVarP(&opts.EventType, "event", "E", "", "Name of the event type to forward")
```

Comment:
Should we make this a `cmd.Flags().StringSliceVarP` so that we can let customers forward multiple events just like the web UI?